### ISSUE TITLE
False negative for specifying non-tuple `Sequence[type[Exception]]` with `except*`

### ISSUE THREAD

**Issue by rsokl** [microsoft/pyright#7483]
`except*` can only be used with `type[Exception] | tuple[type[Exception], ...]`

E.g.  the following raises `TypeError: catching classes that do not inherit from BaseException is not allowed`

```python
try:
    raise OSError('error 1')
except* [OSError, SystemError]:
    ...
```

pyright does not flag this as an error.

**Comment by erictraut** [comment]
It turns out this same issue occurs for `except` (without `*`).

```python
try:
    raise OSError('error 1')
except [OSError, SystemError]:
    ...
```

I was under the wrong impression that the runtime supported arbitrary iterators in this case, but it looks like it specifically requires tuples.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.355, which I just published.
